### PR TITLE
FS-2707-prevent-overwriting-existing-location-data

### PR DESCRIPTION
**Pull request by ramsharma-prog**
This PR adds  logic:

- Checking for existing location data before writing new data to the Assessment records table.
- Preventing the overwriting of existing location data in the Assessment records table.
- Verifying if the existing location data has an error flag set to True. If so, the location data will be rewritten irrespective of any existing location data.

### CODE REVIEW

File: tests/test_db_function.py [hunk 1]
```diff
@@ -228,26 +228,161 @@ def test_update_workflow_status_on_insert(
 
 
 @pytest.mark.apps_to_insert([test_input_data[0]])
-def test_bulk_update_location_data(_db, seed_application_records):
+def test_bulk_update_location_data_with_valid_value(
```
Comment (RamuniN) [inline tests/test_db_function.py:231]: Instead of 3 test cases, can we parametrize the test `test_bulk_update_location_data` with 3 testdata sets ?

Comment (RamuniN) [inline tests/test_db_function.py:231]: Instead of 3 test cases, can we parametrize a single test function with 3 data sets ?

Comment (RamuniN) [inline tests/test_db_function.py:231]: Instead of 3 test cases, can we parametrize the test `test_bulk_update_location_data` with 3 testdata sets ?

Comment (ramsharma-prog) [reply tests/test_db_function.py:231]: Thanks, @RamuniN 
I have parametrized tests

Comment (ramsharma-prog) [reply tests/test_db_function.py:231]: Thanks, @RamuniN 
I have parametrized tests

Comment (ramsharma-prog) [reply tests/test_db_function.py:231]: Thanks, @RamuniN 
I have parametrized tests

**Review by RamuniN — CHANGES_REQUESTED** [review]
LGTM. Please check the tests. It is failing on my machine locally.

**Comment by RamuniN** [comment]
Ignore the previous comment. I think issues with my docker, I cannot verify the unit tests locally

**Review by RamuniN — APPROVED** [review]
:)

**Review by tferns — APPROVED** [review]

File: db/queries/assessment_records/queries.py [hunk 2]
```diff
@@ -363,15 +363,37 @@ def bulk_update_location_jsonb_blob(application_ids_to_location_data):
...
+                stmt,
+                {
+                    "app_id": item["application_id"],
+                    "location_data": item["location"],
+                },
+            )
+
+        else:
+            current_app.logger.info("Location data already exists")
+            continue
```
[the remark this answers is not in the document]
Comment (ramsharma-prog) [reply db/queries/assessment_records/queries.py:395]: I added it because just in case I do not want the loop to break 😆

**Pull request merged**